### PR TITLE
Charts details

### PR DESCRIPTION
**Pull request by pranayairan**
first take on charts details, adding adapter delegates, creating coin details activity and historical chart module.

### CODE REVIEW

File: app/src/main/java/com/binarybricks/coinhood/CoinHoodApplication.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /** A tree which logs important information for crash reporting.  */
+    private class CrashReportingTree : Timber.Tree() {
+        override fun log(priority: Int, tag: String, @NonNull message: String, t: Throwable?) {
+            if (priority == Log.VERBOSE || priority == Log.DEBUG) {
+                return
+            }
+
+//            FakeCrashLibrary.log(priority, tag, message)
```
Comment (pragyagrawal) [inline app/src/main/java/com/binarybricks/coinhood/CoinHoodApplication.kt]: remove if not required

File: app/src/main/java/com/binarybricks/coinhood/network/CoinMarketCapApi.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.rmnivnv.cryptomoon.model.network
+
+/**
+ * Created by rmnivnv on 31/08/2017.
+ */
+interface CoinMarketCapApi {
+
+//    @GET
+//    fun getTopCoins(@Url url: String,  @Query("limit") limit: Int): Single<List<TopCoinData>>
```
Comment (pragyagrawal) [inline app/src/main/java/com/binarybricks/coinhood/network/CoinMarketCapApi.kt]: same here

File: app/src/main/java/com/binarybricks/coinhood/network/CryptoCompareAPI.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+/**
+ * Created by rmnivnv on 11/07/2017.
+ */
+interface CryptoCompareAPI {
+
+    //    @GET
+//    fun getCoinsList(@Url url: String): Single<AllCoinsResponse>
```
Comment (pragyagrawal) [inline app/src/main/java/com/binarybricks/coinhood/network/CryptoCompareAPI.kt]: same

File: app/src/main/java/com/binarybricks/coinhood/network/CryptoCompareAPI.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    fun getPrices(@Query("fsyms") fromSymbol: String, @Query("tsyms") toSymbol: String): Single<JsonObject>
+
+    @GET("{period}")
+    fun getCryptoHistoricalData(@Path("period") period: String,
+                                @Query("fsym") fromCurrencySymbol: String?,
+                                @Query("tsym") toCurrencySymbol: String?,
+                                @Query("limit") limit: Int,
+                                @Query("aggregate") aggregate: Int): Single<CryptoCompareHistoricalResponse>
+
+//    @GET("top/pairs")
```
Comment (pragyagrawal) [inline app/src/main/java/com/binarybricks/coinhood/network/CryptoCompareAPI.kt]: same

File: app/src/main/res/layout/activity_coin_details.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.binarybricks.coinhood.stories.coindetails.CoinDetailsActivity">
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/rvCoinDetails"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginBottom="8dp"
```
Comment (pragyagrawal) [inline app/src/main/res/layout/activity_coin_details.xml]: extract dimens in all xml files

File: app/src/main/java/com/binarybricks/coinhood/CoinHoodApplication.kt (hunk 1 above)
Comment (pranayairan) [reply app/src/main/java/com/binarybricks/coinhood/CoinHoodApplication.kt]: we will use this in future, keeping it for now, in the final version i will remove it. Added a TODO.

File: app/src/main/java/com/binarybricks/coinhood/network/CoinMarketCapApi.kt (hunk 2 above)
Comment (pranayairan) [reply app/src/main/java/com/binarybricks/coinhood/network/CoinMarketCapApi.kt]: all needed, disabled for now will be enabling them in next PR.

**Pull request merged**